### PR TITLE
Apply accessibility updates to case_workers views

### PR DESCRIPTION
**Pull request by naseberry**
#### What
Accessibility and coding standard updates to case_worker views.

#### Ticket
[CBO-784](https://dsdmoj.atlassian.net/browse/CBO-784)

#### Why
To make this section of CCCD accessible for all users.

#### How
- Ensure elements have sufficient color contrast
- Ensure table headers in a data table refers to data cells
- Improve coding standard

### CODE REVIEW

File: app/assets/stylesheets/v1/datatables.min.scss [hunk 1]
```diff
@@ -10,16 +10,1075 @@
...
+table.dataTable {
```
Comment (jsugarman) [inline app/assets/stylesheets/v1/datatables.min.scss]: This does not appear to be minified?!

File: app/views/case_workers/admin/allocations/_allocation.html.haml [hunk 2]
```diff
@@ -1,76 +1,82 @@
...
     .grid-row.js-typeahead
       .column-one-third.js-typeahead
         = label_tag :quantity_to_allocate, t('.no_of_claims'), class: 'form-label-bold'
-        = number_field_tag :quantity_to_allocate, nil, { style: 'width:100%', min: 0, size: 5, class: 'form-control'}
+        = number_field_tag :quantity_to_allocate, nil, { style: 'width: 100%', min: 0, size: 5, class: 'form-control' }
       .column-one-third
         #cc-caseworker
-          = f.label :case_worker_id, t('dictionary.models.case_worker'), for: "allocation_case_worker_id_autocomplete", class: 'form-label-bold'
-          = f.collection_select :case_worker_id, @case_workers, :id, :name, { include_blank: ''.html_safe }, { class: 'form-control fx-autocomplete' , style: 'width:100%'}
+          = label_tag :allocation_case_worker_id, t('dictionary.models.case_worker'), class: 'form-label-bold'
```
Comment (jsugarman) [inline app/views/case_workers/admin/allocations/_allocation.html.haml:39]: **NOT a blocker** 
I remember looking at this with you. You should be able to use `form_with` like this

```
form_with(model: [:case_workers, :admin, @allocation]) do
  = f.hidden_field :tab, params[:tab]
  ....
  = f.label :case_worker_id, t('dictionary.models.case_worker'), class: 'form-label-bold'
  = f.collection_select :case_worker_id, @case_workers, :id, :name, { include_blank: ''.html_safe }, { class: 'form-control fx-autocomplete', style: 'width: 100%' }
```

note the removal of `allocation` from the field name if you do this

an alternative might be:
```
form_with(url: case_workers_admin_allocations_path, method: :get) do
 ```

File: app/views/case_workers/admin/allocations/_allocation.html.haml [hunk 3]
```diff
@@ -1,76 +1,82 @@
...
+    = t('.allocation_queue')
+  %table#dtAllocation
+    %caption.visually-hidden
+      = t('.table_summary')
+      %noscript
+        = t('shared.not_available')
+    %thead
+      %tr
+        %th{ scope: 'col' }
+          = ''
```
Comment (jsugarman) [inline app/views/case_workers/admin/allocations/_allocation.html.haml:70]: **question**: is this useful/necessary/semantic/accessible?

File: app/views/case_workers/admin/allocations/_re_allocation.html.haml [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
 .grid-row
   .column-full
-    = render partial: 'layouts/header', locals: {page_heading: t('.page_heading')}
+    = render partial: 'layouts/header', locals: { page_heading: t('.page_heading') }
 
 .grid-row
   .column-full
     = form_tag case_workers_admin_allocations_path, method: :get do
-      = hidden_field_tag :tab, params[:tab]
+      -# = hidden_field_tag :tab, params[:tab]
```
Comment (jsugarman) [inline app/views/case_workers/admin/allocations/_re_allocation.html.haml]: did you mean to take this out? is it not needed still.

File: app/views/case_workers/admin/allocations/_re_allocation.html.haml [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
+= render partial: 'errors/error_summary', locals: { ep: @allocation, error_summary: t('.error_summary') }
 
 .grid-row
   .column-full
-    = render partial: 'layouts/header', locals: {page_heading: t('.page_heading')}
+    = render partial: 'layouts/header', locals: { page_heading: t('.page_heading') }
 
 .grid-row
   .column-full
     = form_tag case_workers_admin_allocations_path, method: :get do
```
Comment (jsugarman) [inline app/views/case_workers/admin/allocations/_re_allocation.html.haml:12]: **NOT a blocker** but could try to sort out the nested form issue here as mentioned here [../allocation.html.haml](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/2994/files#r328094880) and use `form_with` too

File: app/views/shared/_determinations_form.html.haml [hunk 6]
```diff
@@ -50,47 +51,51 @@
...
-          %a#refused_reason
-          %legend.bold-normal.form-label#checkbox-control-legend-2
+          %legend#checkbox-control-legend-2.bold-normal.form-label
             = t('.reason_for_refusal')
 
+          %a#refused_reason
           = collection_check_boxes(nil, :state_reason, ClaimStateTransitionReason.refuse_reasons_for(@claim), :code, :description) do |b|
             .multiple-choice
               = b.check_box(checked: params[:state_reason].present? && params[:state_reason].include?(b.value), 'aria-labelledby': "checkbox-control-legend-2 #{b.text.to_css_class}")
-              = b.label(id: b.text.to_css_class) { b.text }
```
Comment (jsugarman) [inline app/views/shared/_determinations_form.html.haml:86]: I don't recall why this was used but it is common and i'd be wary of removing. Feature tests may not cover it sufficiently. Have you checked feature tests cover it and/or manually tested this determination reasons feature.

File: app/views/case_workers/admin/allocations/_allocation.html.haml (hunk 2 above)
Comment (naseberry) [reply app/views/case_workers/admin/allocations/_allocation.html.haml:39]: 👍 , I will update all forms in a later sprint to use `form_with`

**Review by jsugarman — COMMENTED** [review]
great tidy up. One blocking issue is whether that `datatables.min.scss` should be minified?

A few comments/questions, particularly on the determintion reasons check box ids, but otherwise looks good 👍

File: app/views/case_workers/admin/allocations/_allocation.html.haml (hunk 3 above)
Comment (naseberry) [reply app/views/case_workers/admin/allocations/_allocation.html.haml:70]: In this instance, none of the above as our datatable plugin replaces this with within a 'select all' checkbox.

It however made the code more readable to me than just having the `%th`

File: app/views/case_workers/admin/allocations/_re_allocation.html.haml (hunk 5 above)
Comment (naseberry) [reply app/views/case_workers/admin/allocations/_re_allocation.html.haml:12]: As discussed, this instance is not nested

File: app/views/shared/_determinations_form.html.haml (hunk 6 above)
Comment (naseberry) [reply app/views/shared/_determinations_form.html.haml:86]: Good point, I have test and everything seems ok

File: app/views/case_workers/admin/allocations/_allocation.html.haml (hunk 2 above)
Comment (jsugarman) [reply app/views/case_workers/admin/allocations/_allocation.html.haml:39]: 👍

File: app/views/case_workers/admin/allocations/_re_allocation.html.haml (hunk 5 above)
Comment (jsugarman) [reply app/views/case_workers/admin/allocations/_re_allocation.html.haml:12]: 👍

**Review by jsugarman — APPROVED** [review]
👍 looks good. have tested the rejection/refusal and reallocation features working locally as expected too

**Pull request merged**